### PR TITLE
[OpAttr]Adapt tensor output_size for conv2d_transpose and depthwise_conv2d_transpose

### PR DESCRIPTION
**Pull request by 0x45f**
### PR types

Others
### PR changes

Others
### Describe

[OpAttr]Adapt tensor output_size for conv2d_transpose

### CODE REVIEW

File: paddle/phi/infermeta/binary.cc [hunk 1]
```diff
@@ -800,6 +800,38 @@ void ConvTransposeInferMeta(const MetaTensor& x,
...
+                              const std::vector<int>& output_padding,
+                              const IntArray& output_size,
+                              const std::string& padding_algorithm,
+                              int groups,
+                              const std::vector<int>& dilations,
+                              const std::string& data_format,
+                              MetaTensor* out,
+                              MetaConfig config) {
+  std::vector<int32_t> vec_output_size;
+  vec_output_size.reserve(output_size.size());
```
Comment (Aurelius84) [inline paddle/phi/infermeta/binary.cc]: 可以直接：std::vector<int32_t> vec_output_size(output_size.GetData().begin(), output_size.GetData().end());

File: paddle/phi/infermeta/binary.cc [hunk 2]
```diff
@@ -800,6 +800,38 @@ void ConvTransposeInferMeta(const MetaTensor& x,
...
+  std::for_each(
+      output_size.GetData().begin(),
+      output_size.GetData().end(),
+      [&vec_output_size](const int64_t& t) { vec_output_size.push_back(t); });
+  ConvTransposeInferMeta(x,
+                         filter,
+                         strides,
+                         paddings,
+                         output_padding,
+                         vec_output_size,
```
Comment (Aurelius84) [inline paddle/phi/infermeta/binary.cc:822]: 这里不需要区分编译期和执行期么？

File: paddle/phi/infermeta/binary.cc (hunk 1 above)
Comment (0x45f) [reply paddle/phi/infermeta/binary.cc]: 已修改，感谢

File: paddle/phi/infermeta/binary.cc (hunk 2 above)
Comment (0x45f) [reply paddle/phi/infermeta/binary.cc:822]: ConvTransposeInferMeta函数中有区分编译期和执行期

**Review by Aurelius84 — APPROVED** [review]
LGTM

**Review by XiaoguangHu01 — APPROVED** [review]
LGTM

**Pull request merged**